### ISSUE TITLE
Language server does not load dependencies with a target option

### ISSUE THREAD

**Issue by th0mas** [elixir-lsp/elixir-ls#329]
### Environment

* Elixir & Erlang versions (elixir --version): Erlang/OTP 23 
* Operating system: Elixir 1.10.3
* Editor or IDE name (e.g. Emacs/VSCode): VSCode
* Editor Plugin/LSP Client name: ElixirLS

### Troubleshooting

- [x] Restart your editor (which will restart ElixirLS) sometimes fixes issues
- [x] Stop your editor, remove the entire `.elixir_ls` directory, then restart your editor
  * NOTE: This will cause you to have to re-run the dialyzer build for your project

If you're experiencing high CPU usage, it is most likely Dialyzer building the PLTs; after it's done the CPU usage should go back to normal. You could also disable Dialyzer in the settings.

### Issue

When specifying dependencies with a `:target` option, the language server will throw an error stating `module is not loaded and could not be found`

For example:

Declaring a dependency though   
```elixir
{:scenic, "~> 0.10", targets: [:host, :rpi3]},
```
in a Nerves project means that the language server does not recognise `Scenic` in the rest of the project.

Compiling works fine (as expected) and AFAIK `Mix.target()` returns the correct value in development, `:host`.

You can clone https://github.com/dwyl/smart-home-firmware to see this, go to `lib/smart_home_firmware/display/scene/welcome.ex`, but if you want a more minimal example, let me know and I'll put one together.

**Comment by jjcarstens** [comment]
👋 Hey @th0mas! 

In ElixirLS v0.5.0, the ability to set `MIX_TARGET` was added for this specific reason of dealing with targets, mostly in Nerves. It accidentally got overlooked in the latest release changelog, but I'm adding it back now.

I cloned your project and set the target to `rpi3` in the project/folder settings, reloaded ElixirLS and then checked the `welcome.ex` module to see the warning was gone.
![image](https://user-images.githubusercontent.com/11321326/88429774-7a6d1d00-cdb4-11ea-85fa-bb38af77ab1e.png)

The downside is this LS setting is singular - If you have different modules around your project that all require different targets, then you might still see the same issue. Though, that's not something I've every seen and `:targets` mostly just affects which underlying nerves system gets used for compilation
